homeassistant/components/light/__init__.py

```diff
@@ -1062,43 +1063,46 @@ def state_attributes(self) -> dict[str, Any] | None:
         """Return state attributes."""
         data: dict[str, Any] = {}
         supported_features = self.supported_features_compat
-        supported_color_modes = self._light_internal_supported_color_modes
+        supported_color_modes = self.supported_color_modes
+        legacy_supported_color_modes = (
+            supported_color_modes or self._light_internal_supported_color_modes
+        )
```

Comment:
This is not a change, just renamed the variable for clarity since its confusing that we have both in this function